instrumentation/all/opentelemetry-instrumentation-all.gemspec

```diff
@@ -26,6 +26,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
   spec.required_ruby_version = '>= 2.5.0'
 
+  spec.add_dependency 'opentelemetry-instrumentation-action_view', '~> 1.0.0.rc1'
```

Comment:
This version constraint from `inst-all` to `inst-action_view` uses a version number later than [`inst-action_view`'s current version](https://github.com/open-telemetry/opentelemetry-ruby/blob/97e67bf59a7c379232f6d8d827b542ec5c2ecaba/instrumentation/action_view/lib/opentelemetry/instrumentation/action_view/version.rb#L10) (0.19.0).

Found attempting to use this branch for smoke testing.

<details><summary>Unhappy Bundler</summary>

```
Resolving dependencies...
Bundler could not find compatible versions for gem "opentelemetry-instrumentation-action_view":
  In Gemfile:
    opentelemetry-instrumentation-all was resolved to 0.18.0, which depends on
      opentelemetry-instrumentation-action_view (~> 1.0.0.rc1)

Could not find gem 'opentelemetry-instrumentation-action_view (~> 1.0.0.rc1)',
  which is required by gem 'opentelemetry-instrumentation-all', in any of the sources.
```

</details>

Reply:
I think this is because the instrumentation generator uses the api/sdk version as a default, when that might not actually be what anyone wants now that they are not kept in lockstep. cc @robertlaurin

Reply:
@robbkidd I [updated](https://github.com/open-telemetry/opentelemetry-ruby/pull/796/commits/23b8c55e408a8f359db35c3ad455de84f8b18544) to the correct version, but since the gem is unreleased I don't know how bundler will actually behave with the `-all` gem. When I was testing locally I had just installed the gem from a `file` path in Bundler.

Reply:
I am very interested in your smoke-testing though; this "works on my machine" ™️ but I would not be at all surprised if it breaks horribly in ways I haven't foreseen.

Reply:
I don't think we actually talked about what the new default should be but lets just start by setting it to something that makes more sense than the api version https://github.com/open-telemetry/opentelemetry-ruby/pull/803

Reply:
I rubbed a lot of `gem: _, git: _, branch: _, glob: _` into my Gemfile which got Bundler to resolve them at install time OK, but I'm still having issues with library load at runtime. So I'm cloning your fork and trying it locally with `file: _` now.

Reply:
@robbkidd did this ever work for you?

Reply:
I ... think yes? I'll refresh mental state on it and report back tomorrow!

Reply:
Succeeded in using the new gem, yes.

```ruby
gem 'opentelemetry-api'
gem 'opentelemetry-sdk'
gem 'opentelemetry-exporter-otlp'

github 'ahayworth/opentelemetry-ruby', branch: 'ahayworth-activesupport-notifications', glob: 'instrumentation/*/*.gemspec' do
  gem 'opentelemetry-instrumentation-action_view'
  gem 'opentelemetry-instrumentation-rails'
end
```

Currently seeing errors and no spans produced for ActionView renderings:

```
E, [2021-06-26T00:11:38.603174 #1] ERROR -- : OpenTelemetry error: calls to detach should match corresponding calls to attach.
E, [2021-06-26T00:11:38.603274 #1] ERROR -- : OpenTelemetry error: calls to detach should match corresponding calls to attach.
E, [2021-06-26T00:11:38.603454 #1] ERROR -- : OpenTelemetry error: calls to detach should match corresponding calls to attach.
```

Of course, I could be doing something wrong.

Reply:
@ahayworth With some judicious `puts` statements sprinkled in `SpanSubscriber`'s `start()` and `finish()`, I'm seeing that the `payload` does not have the [span or token](https://github.com/open-telemetry/opentelemetry-ruby/pull/796/files#diff-e29d49095a6b1901328545a9e5571958045335dfd677018d2c1c4e99c53ee886R29-R30) tucked into it. So, I'm thinking [the OTel replacement for Fanout](https://github.com/open-telemetry/opentelemetry-ruby/pull/796/files#diff-f637f95bc8fb1aad5c059ae044b926502075c6797033e81ff7b18c713ad4c0b4R15) isn't taking effect, maybe?

Reply:
> So, I'm thinking the OTel replacement for Fanout isn't taking effect, maybe?

@robbkidd Quite probably, yes. Can you share how you're initializing the OpenTelemetry SDK in your tests? (And/or if you have any kind of public test app that reproduces it, that'd be fantastic too)

Reply:
@ahayworth My test lab is very messy which could very well be the origin of my problem. I'll work on getting it into a state that is shareable.